plrustc/plrustc/src/lints.rs

```diff
@@ -107,6 +107,65 @@ impl<'tcx> LateLintPass<'tcx> for PlrustAutoTraitImpls {
     }
 }
 
+declare_plrust_lint!(
+    pub(crate) PLRUST_SUSPICIOUS_TRAIT_OBJECT,
+    "Disallow suspicious generic use of trait objects",
+);
+
+declare_lint_pass!(PlrustSuspiciousTraitObject => [PLRUST_SUSPICIOUS_TRAIT_OBJECT]);
+
+impl<'tcx> LateLintPass<'tcx> for PlrustSuspiciousTraitObject {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &hir::Expr) {
+        let path_segments = match &expr.kind {
+            hir::ExprKind::Path(hir::QPath::Resolved(_, path)) => path.segments,
+            hir::ExprKind::Path(hir::QPath::TypeRelative(_, segment, ..))
+            | hir::ExprKind::MethodCall(segment, ..) => std::slice::from_ref(*segment),
+            _ => return,
```

Comment:
For base-case ignores like this on enums with several dozen variants, when it's not easy to exhaustively match, can we comment on the condition the rest of the other variants fulfill that allows us to ignore them? I'm thinking about refactors to ExprKind that might force us to reevaluate that, and having the assumption clearly stated can help.

It also makes it easier to evaluate the rest of the code because it feeds forward into a better understanding of "what are we looking for" (by specifying what we're not looking for).

Reply:
I don't really know what you mean here.